code/src/components/Labyrinth.js

```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import GameOver from "./GameOver";
+import StartPage from './StartPage';
+import GameScreen from "./GameScreen";
+import Loading from "./Loading";
+
+import { Container } from "./Styles";
+
+const Labyrinth = () => {
+    const stage = useSelector(store => store.labyrinth.stage);
+    const isLoading = useSelector(store => store.labyrinth.isLoading);
+
+    return (
+        <>
+            {isLoading && <Loading />}
+            {!isLoading && (
+                <Container>
+                    {stage 

```

Comment:
Easy to read and follow. Very great solution on using two ternary operators for different scenarios; startPage, gameScreen and GameOver.